monitor/main.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir
+
+// Monitor is not an example program on its own, but an accompanying program
+// that designed to poll the cluster and save the status to disk. The monitor
+// program will exit if the status of the cluster can not be determined.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/base"
+	"github.com/cockroachdb/cockroach/security"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/log"
+	"github.com/cockroachdb/cockroach/util/retry"
+)
+
+var outputDir = flag.String("dir", ".", "Directory in which to output the status logs to.")
+var interval = flag.Duration("interval", 10*time.Second, "Interval in which to pole the cluster's status.")
+var addr = flag.String("addr", ":26257", "The host:port of the cockroach cluster.")
+var insecure = flag.Bool("insecure", false, "True if using an insecure connection.")
+var user = flag.String("user", security.RootUser, "User used to connect to the cluster.")
+var certs = flag.String("certs", "certs", "Directory containing RSA key and x509 certs. This flag is required if --insecure=false.")
+var endpoint = flag.String("endpoint", "_status/nodes", "Status endpoint to monitor and log.")
+
+var retryOptions = retry.Options{
+	InitialBackoff: 100 * time.Millisecond,
+	MaxRetries:     4,
+	Multiplier:     2,
+}
+
+// request returns the result of performing a http get request.
+func request(url string, httpClient *http.Client) ([]byte, bool) {
+	for r := retry.Start(retryOptions); r.Next(); {
```

Comment:
I disagree.  
I explicitly wanted to ensure there was some retry logic in case there are any weird temporary network issues.  I was actually thinking of increasing the retries to around 10 or 11. Actually, I'm going to add that.